abi/serializer_test.go

```diff
@@ -0,0 +1,396 @@
+package abi
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
```

Comment:
I would like to see at least one complex encoding/decoding with struct in struct with all kind of lists, numbers, enums and stuff. This can be a TODO for another time, no problem :D

Reply:
Agree, can be an integration test

Reply:
Indeed. Added some real-world scenarios based on multisig.